addons/pos_sale/models/pos_order.py

```diff
@@ -71,14 +71,17 @@ def create_from_ui(self, orders, draft=False):
             # track the waiting pickings
             waiting_picking_ids = set()
             for so_line in so_lines:
-                for stock_move in so_line.move_ids.group_id.stock_move_ids:
+                for stock_move in so_line.move_ids:
```

Comment:
I notice that we are undoing what is done here: https://github.com/odoo/odoo/pull/109515

Have you take into account the issue described there?

Reply:
Yes, that's the purpose of the loop I added. It updates the value for the other steps when using multi steps delivery